journal/src/main/java/io/camunda/zeebe/journal/file/JournalSegments.java

```diff
@@ -0,0 +1,388 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.journal.file;
+
+import io.camunda.zeebe.journal.JournalException;
+import io.camunda.zeebe.journal.file.record.CorruptedLogException;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.DirectoryStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.NavigableMap;
+import java.util.SortedMap;
+import java.util.concurrent.ConcurrentSkipListMap;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class JournalSegments {
```

Comment:
It is not actually thread-safe. We expect that only there is only writer. But we don't enforce it. We have rwlocks only to synchronize between readers and the writer in specific operations. `append` for example doesn't use any synchronization because we assume that there is only one writer. We should either document this or enforce it as part of other issue. For now, it doesn't make sense to me that some methods are "synchronized" while some are not. That's why I removed it.